sktime_dl/utils/_models.py

```diff
@@ -25,3 +27,45 @@ def save_trained_model(model, model_save_directory, model_name, save_format='h5'
             file_name = model_name + '.hdf5'
         path = Path(model_save_directory) / file_name
         model.save(path)  # Add save_format here upon migration from keras to tf.keras
+
+
+def check_is_fitted(estimator, msg=None):
```

Comment:
Why don't we make this a method on the base class? So that we can simply call 
```python
estimator.check_is_fitted()
```

Or inside other methods:
```python
self.check_is_fitted()
```

Reply:
@Withington any thoughts on this? In this end, I'd prefer to have a common way of checking this across all estimators including base sktime.

Reply:
So I agree putting it somewhere else in general, question being where

There's: 

1.  In the particular estimator implementations (e.g. MLPClassifier), in sktime-dl
2.  In the BaseDeepClassifier/BaseDeepRegressor/(in future Base forecaster presumably, etc) classes, in sktime-dl
3.  In the BaseClassifier/BaseRegressor/etc classes, in sktime
4.  Ultimately in the BaseEstimator class, in sklearn

Can't do 4 directly, so we either duplicate the code at some level to put it into each Base[Task] or BaseDeep[Task] class, OR have another class in sktime that Base[Task] inherits from alongside BaseEstimator that includes utility code we wish we could just add to BaseEstimator directly, OR have the current setup of it being an external utility

Ultimately the difference in usage is self.check_is_fitted vs check_is_fitted(self), though agree the first makes more immediate sense/looks nicer

Reply:
This version of 4 "have another class in sktime that Base[Task] inherits from alongside BaseEstimator that includes utility code we wish we could just add to BaseEstimator directly" seems fitting. 

Or, similarly, 
5. have another class in sktime that inherits from BaseEstimator and adds in extra functionality such as check_is_fitted.  Base[Task] then inherits from this new class instead of BaseEstimator.

Reply:
I think 5. makes sense, but it may take a bit longer until this ends up in a release in sktime, so I suggest we implement it here in the mean time. Once I have time to implement this in sktime, we can remove it here. Sounds good?

Reply:
Sounds like a good plan, yes.

Reply:
Actually, it's not that easy unfortunately, because we decided to implement `BaseRegressor` and `BaseClassifier` in sktime rather than using sklearn's mixin classes, so we would have to rewrite a lot more, so the best is to wait for this to go into sktime, I've added it already on my forecast branch, and I'll open a new issue here to remind us.